src/app/(contacts)/(components)/DeleteContactDialogBtn.tsx

```diff
@@ -0,0 +1,26 @@
+'use client';
+
+import { toast } from 'sonner';
+import { deleteContactAction } from '../actions';
+
+type Props = {
+  id: string;
+  onSuccess?: () => void;
+};
+
+const DeleteContactDialogBtn = ({ id, onSuccess }: Props) => {
+  const handleClick = async () => {
+    try {
+      await deleteContactAction(id);
+      toast.success('Contact deleted successfully');
+      onSuccess?.();
+    } catch (error) {
+      toast.error('Failed to delete contact');
+      console.error('Failed to delete contact:', error);
+    }
+  };
+
+  return <button onClick={handleClick}>Delete</button>;
```

Comment:
jeżeli ten handler jak async to warto wrzucić na button loader albo disabled, żeby uniknąć kolejnych wywołań.